packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js

```diff
@@ -33,14 +41,17 @@ export async function makeWallet(
     return JSON.stringify([...inboxState.values()]);
   }
 
-  async function updatePursesState(purseName, purse) {
-    const balance = await E(purse).getBalance();
-    const {
+  async function updatePursesState(pursePetname, purse) {
+    const { extent } = await E(purse).getBalance();
+    const assay = await E(purse).getAssay();
+    const assayPetname = assayToPetname.get(assay);
```

Comment:
Can you instead save a round trip and do:
```js
const [{ extent }, assay] = await Promise.all([
  E(purse).getBalance(),
  E(purse).getAssay(),
]);
```

Reply:
Yup, good idea!